src/v/cloud_storage/materialized_segments.cc

```diff
@@ -310,8 +314,19 @@ void materialized_segments::maybe_trim_segment(
         // this will delete and unlink the object from
         // _materialized collection
         if (st.parent) {
-            to_offload.push_back(
-              std::make_pair(st.parent.get(), st.base_rp_offset()));
+            if (
+              st.parent->state()
+              == remote_partition::partition_state::running) {
+                to_offload.push_back(
+                  std::make_pair(st.parent.get(), st.base_rp_offset()));
+            } else {
+                vlog(
+                  cst_log.info,
+                  "Materialized segment {} offset {} not offloaded, partition "
+                  "is stopping",
+                  st.ntp(),
+                  st.base_rp_offset());
+            }
```

Comment:
Just want to confirm: it's only important that the offload is skipped since that's what iterates over the `_segments` map, right?

Would it make sense to check the state of the `remote_partition` during the call to `offset_segment()`?

Reply:
Yes, that would also work as right now the end result is the same. However if we avoid processing the segment by not adding it to the list, it will protect us from future evolution of the code where we do something else with the contents of the offload list later in the code, potentially trying to change segment state etc. 

The intention behind this check is that as the remote partition is already stopping, `materialized_segments` should leave any segments from that partition alone because all of the segments will soon be stopped, if that makes sense.